balloc/balloc.c

```diff
@@ -861,9 +884,9 @@ static int balloc_groups_write(struct m0_balloc *bal)
 	  by this parameter.
    @return 0 means success. Otherwise, error number will be returned.
  */
-static int balloc_format(struct m0_balloc *bal,
-			 struct m0_balloc_format_req *req,
-			 struct m0_sm_group *grp)
+static int balloc_format(struct m0_balloc               *bal,
+			 struct m0_ad_balloc_format_req *req,
+			 struct m0_sm_group             *grp)
```

Comment:
I gave different option here.
"struct m0_balloc_format_req *req" is changed to "struct m0_ad_balloc_format_req *req".
But, the "struct m0_ad_balloc_format_req" is AD specific.

"struct m0_balloc_format_req" is more general and can be used by Linux stob too.
Maybe we can still use the same struct, but add some more fields to "struct m0_balloc_format_req"?

Reply:
Had earlier tried around this, Faced the compilation header circular dependency issue.  To achieve this, we need to move m0_ad_balloc_format_req ad  as m0_balloc_format_req in balloc.h. then faced  header dependency issue.  Then latter drops that thought process as I felt balloc is more specific to AD.
Please correct me here, i may wrong here.  For Linux Stob, balloc has no role. So balloc_format_req is more specific to AD, on this thought process I used it only for AD.

Reply:
OK.